packages/celotool/src/e2e-tests/slashing_tests.ts

```diff
@@ -0,0 +1,249 @@
+// tslint:disable-next-line: no-reference (Required to make this work w/ ts-node)
+/// <reference path="../../../contractkit/types/web3.d.ts" />
+
+import { ContractKit, newKitFromWeb3 } from '@celo/contractkit'
+import { NULL_ADDRESS } from '@celo/utils/lib/address'
+import BigNumber from 'bignumber.js'
+import { assert } from 'chai'
+import * as rlp from 'rlp'
+import Web3 from 'web3'
+import { getContext, GethTestConfig, sleep } from './utils'
+
+function headerArray(web3: Web3, block: any) {
+  return [
+    block.parentHash,
+    block.sha3Uncles,
+    block.miner,
+    block.stateRoot,
+    block.transactionsRoot,
+    block.receiptsRoot,
+    block.logsBloom,
+    web3.utils.toHex(block.difficulty),
+    block.number,
+    block.gasLimit,
+    block.gasUsed,
+    block.timestamp,
+    block.extraData,
+    block.mixHash,
+    block.nonce,
+  ]
+}
+
+function headerFromBlock(web3: Web3, block: any) {
+  return rlp.encode(headerArray(web3, block))
+}
+
+describe('slashing tests', function(this: any) {
+  const gethConfig: GethTestConfig = {
+    migrate: true,
+    instances: [
+      // Validators 0 and 1 are swapped in and out of the group.
```

Comment:
Remove these comments here and below, (and replace with accurate descriptions if necessary)